browser/src/control/Control.UIManager.js

```diff
@@ -658,8 +659,21 @@ L.Control.UIManager = L.Control.extend({
 	showButton: function(buttonId, show) {
 		if (!this.notebookbar)
 			this.showButtonInClassicToolbar(buttonId, show);
-		else
+		else {
 			this.notebookbar.showShortcutsButton(buttonId, show);
```

Comment:
btw: didi you look at that `showShortcutsButton` function?
it does `this.reloadShortcutsBar();` inside it should already do the show/hide logic.

But I see my mistake there: `this.hiddenShortcutButtons.push(buttonId);` is called unconditionaly, even if `show == true`. Maybe that is enough to fix and it will work?

Reply:
and maybe we don't need that `hiddenShortcutButtons` inside notebookbar at all, as now you have this "global" `hiddenButtons` thing...

Reply:
I have now removed `showShortcutsButton` and just call `reloadShortcutsBar()` and use `UIManager.hiddenButtons`